dask/utils.py

```diff
@@ -1124,12 +1124,23 @@ def is_arraylike(x):
     False
     """
     from .base import is_dask_collection
+    from .array.utils import IS_NEP18_ACTIVE
+
+    # Fallback to True for cases where NEP-18 isn't active (e.g. older versions of NumPy)
+    implements_array_fuction = (
+        hasattr(x, "__array_function__") if IS_NEP18_ACTIVE else True
+    )
```

Comment:
This doesn't totally work as we can't distinguish between "duck" and "input" arrays for all NumPy versions we support just using `__array_function__`. Thoughts on using `hasattr(x, "__array_function__") or hasattr(x, "__array_ufunc__")` instead?

Reply:
Seems reasonable to me, unless there are non-duck arrays that also implement `__array_ufunc__`, I'm not sure if that's the case but it's worth a try.

Reply:
This has been updated below. Noted in this review comment ( https://github.com/dask/dask/pull/7321#discussion_r587917035 ). Would be good to hear what people think of this formulation